Adding SR570 current amplifier driver

pyMatJ opened the pull request:
This adds the driver for the SR570 current amplifier (from Stanford Research Systems).
The SR570 communicates through a RS232 listen-only so it is up to the end user to keep track of the values of the set parameters, no function in the driver stores or returns a value (see https://github.com/pymeasure/pymeasure/issues/387 moved to https://github.com/pymeasure/pymeasure/discussions/406)

CasperSchippers reviewed (changes requested):
Thanks for the addition.

Could you also create the appropriate doc-files (docs/api/instruments/srs/sr570.rst, you can take an example from the other instruments) and add it to the index files (docs/api/instruments/srs/index.rst). Finally, don't forget to add yourself to the two authors lists (AUTHORS.txt and docs/about/authors.txt)

pyMatJ commented:
I tried to make all these modifications. 
- the file terminates on an empty line
- the init is at the beginning of the class 
- I added the allowed values in the docstring
- I added setting properties in addition to wrapper functions and used mapping values when recommended (it is indeed much more easy to read like this)

Thanks for your inputs !

CasperSchippers commented on pymeasure/instruments/srs/sr570.py:
```diff
@@ -0,0 +1,214 @@
+    # Methods        #
+    ####################
+    
+    def enable_bias(self):
+        """Turns the bias voltage on"""
+        self.write("BSON 1")
+    
+    def disable_bias(self):
+        """Turns the bias voltage off"""
+        self.write("BSON 0")
```
As you have implemented the `bias_enabled` property, maybe you could change `self.write("BSON 1")` to `self.bias_enabled = True` (same holds for the other wrappers below)

pyMatJ commented on docs/api/instruments/srs/index.rst:
```diff
@@ -9,5 +9,6 @@ This section contains specific documentation on the Stanford Research Systems (S
 .. toctree::
    :maxdepth: 2
 
+   sr570
```
ah sorry, it got lost in a commit -a that does not add files not previously indexed... I'm a bit new to this, my bad.

CasperSchippers replied:
Ah, yes; no worries

CasperSchippers commented on pymeasure/instruments/srs/sr570.py:
```diff
@@ -0,0 +1,214 @@
+        """Resets the amplifier to default settings"""
+        self.write('*RST')
+    
+    def blank_front(self):
```
Could you do the same for this as for the other wrappers with the property?

CasperSchippers commented on pymeasure/instruments/srs/sr570.py:
```diff
@@ -0,0 +1,214 @@
+        """"Reset the filter capacitors to clear an overload condition"""
+        self.write("ROLD")
+    
+    def reset_instrument(self):
+        """Resets the amplifier to default settings"""
+        self.write('*RST')
```
A method for sending `*RST` is standard implemented (named `reset`), so you can remove this to prevent code duplication.

pyMatJ replied:
Ah yes I see, I did not check this. This is fixed now, thanks for your thorough reviewing work :)

CasperSchippers reviewed (commented):
Sorry, i missed this one thing last time i checked; after this it can be merged I think.
